### PR TITLE
PIMS-1623/1622: Handling project notifications and tasks

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
## 🎯 Summary

  
[PIMS-1623](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-1623)
[PIMS-1622](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-1622)

* Updated the project notification handler with correct logic for calling the notification service to fire emails when a project status update happens.
* Updated project task handler logic slightly.
* Implemented a bunch of custom frontend logic to handle showing and selecting tasks for a given status transition when editing the general info section.
* Revamped the documentation data card section to show a list of tasks the project has gone through previously, grouped by status.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

**Review by LawrenceLau2020 — COMMENTED** [review]
I think we're missing the UI to display the different notes: agency, reporting, shared and private notes.

**Comment by GrahamS-Quartech** [comment]
> I think we're missing the UI to display the different notes: agency, reporting, shared and private notes.

Yeah, this will just have to be another task kinda like the agency interest responses. Bummer that we sorta missed these in the initial design phase.

File: react-app/src/pages/Forbidden.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Box, Paper, Typography } from '@mui/material';
+import React from 'react';
+
+const Forbidden = () => {
```
Comment (dbarkowsky) [inline react-app/src/pages/Forbidden.tsx:4]: Would be good for redirections through the AuthGuard.

**Review by dbarkowsky — COMMENTED** [review]
image
I like that we can see all these now. Might have to confirm if those duplicate Appraise/Approve tasks are needed. We also probably won't show all of these at once in the future, as some projects will never need ERP exemption tasks, etc.

File: react-app/src/pages/Forbidden.tsx (hunk 1 above)
Comment (GrahamS-Quartech) [reply react-app/src/pages/Forbidden.tsx:4]: Comitted this by accident... A leftover from when I was trying to do exactly that. We probably actually will want this eventually so I guess I won't get rid of it.

File: express-api/src/services/projects/projectsServices.ts [hunk 2]
```diff
@@ -343,7 +351,8 @@ const handleProjectTasks = async (
         ProjectId: project.Id,
         CreatedById: existingTask ? existingTask.CreatedById : project.CreatedById,
         UpdatedById: existingTask ? project.UpdatedById : undefined,
-        CompletedOn: task.IsCompleted ? new Date() : undefined,
+        IsCompleted: task.IsCompleted,
+        CompletedOn: !existingTask && task.IsCompleted ? new Date() : undefined,
```
[the remark this answers is not in the document]
Comment (GrahamS-Quartech) [reply express-api/src/services/projects/projectsServices.ts:355]: Did `CompletedOn: !existingTask?.CompletedOn && task.IsCompleted ? new Date() : undefined,` instead.

**Review by dbarkowsky — APPROVED** [review]
Happy with that last change.

**Pull request merged**